subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -155,33 +160,40 @@ static int key_gen_public_key(struct bt_conn *conn, struct fp_keys_keygen_params
 	return err;
 }
 
-static int key_gen_account_key(struct bt_conn *conn, struct fp_keys_keygen_params *keygen_params)
+static bool key_gen_account_key_check(const uint8_t *account_key, void *context)
 {
 	int err;
+	uint8_t req[FP_CRYPTO_AES128_BLOCK_LEN];
+	struct bt_conn *conn =
```

Comment:
I agree with @MarekPieta

Reply:
Sure, decided to use hybrid approach in order to avoid the following syntax:

```
       err = ak_check_context->keygen_params->req_validate_cb(ak_check_context->conn,
                                                              req,
                                                              ak_check_context->keygen_params->context);
```